content/Constants/Constants.q.md

```diff
@@ -0,0 +1,1453 @@
+
+:::
+
+
+/// type=CR, answer=[tests/Constants/RemoveVisibilityKeywordOnConstantDeclarationTest.php], filename=[CircularShape.php,Circle.php]
+
+Correct the code so that it outputs the strings `Radius: 3.5`, `Diameter: 12.25`, `Area: 38.4846`, and `Circumference: 21.9912` in separate lines.
+
+```php
+<?php
+abstract class CircularShape
+{
+    public const PI = 3.1416;
+    private $radius;
+    
+    public function __construct($radius)
+    {
+        $this->radius = $radius;
+    }
+    
+    public function getRadius()
+    {
+        return $this->radius;
+    }
+    
+    abstract public function diameter();
+    abstract public function area();
+    abstract public function circumference();
+}
+?>
+```
+
+```php
+<?php
+require_once('./CircularShape.php');
+class Circle extends CircularShape
+{
+    public function area()
+    {
+        return self::PI * $this->getRadius() * $this->getRadius();
+    }
+    
+    public function diameter()
+    {
+        return $this->getRadius() * $this->getRadius();
+    }
+    
+    public function circumference()
+    {
+        return 2 * self::PI * $this->getRadius();
+    }
+    
+    public function display()
+    {
+        echo "Radius: " . $this->getRadius();
+        echo "\nDiameter: " . $this->diameter();
+        echo "\nArea: " . $this->area();
+        echo "\nCircumference: " . $this->circumference(); 
+    }
+}
+
+$circleObject = new Circle(3.5);
+$circleObject->display();
+?>
+```
+
+
+:::
+
+/// type=REPL, filename=[CircularShape.php,Circle.php]
+
+```php
+<?php
+abstract class CircularShape
+{
+    const PI = 3.1416;
+    private $radius;
+    
+    public function __construct($radius)
+    {
+        $this->radius = $radius;
+    }
+    
+    public function getRadius()
+    {
+        return $this->radius;
+    }
+    
+    abstract public function diameter();
+    abstract public function area();
+    abstract public function circumference();
+}
+?>
+```
+
+```php
+<?php
+require_once('./CircularShape.php');
+class Circle extends CircularShape
+{
+    public function area()
+    {
+        return self::PI * $this->getRadius() * $this->getRadius();
+    }
+    
+    public function diameter()
+    {
+        return $this->getRadius() * $this->getRadius();
+    }
+    
+    public function circumference()
+    {
+        return 2 * self::PI * $this->getRadius();
+    }
+    
+    public function display()
+    {
+        echo "Radius: " . $this->getRadius();
+        echo "\nDiameter: " . $this->diameter();
+        echo "\nArea: " . $this->area();
+        echo "\nCircumference: " . $this->circumference(); 
+    }
+}
+
+$circleObject = new Circle(3.5);
+$circleObject->display();
+?>
+```
+/// type=SS, answer=[1]
+
+In the statement `return 2 * self::PI * $this->getRadius();` on line 17 of `Circle.php`, replace `self::PI` with `$this->PI`. Execute the program. What is the error message?
```

Comment:
Grammarly suggested this correction:

statement,

"It appears that you are missing a comma after the introductory phrase **In the statement**.
Consider adding a comma."